ci-operator/step-registry/storage/create/csi-aws-efs/cross-account/storage-create-csi-aws-efs-cross-account-commands.sh

```diff
@@ -0,0 +1,315 @@
+                "AWS": "arn:aws:iam::${AWS_ACCOUNT_A_ID}:root"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+EOF
+
+
+
+ACCOUNT_B_ROLE_ARN=$(aws iam create-role \
+  --role-name "${CLUSTER_NAME}-cross-account-aws-efs-csi-operator" \
+  --assume-role-policy-document file://"${SHARED_DIR}"/AssumeRolePolicyInAccountB.json \
+  --query "Role.Arn" --output text)
+logger "INFO" "Created efs csi driver operator role ${ACCOUNT_B_ROLE_ARN} and trust policy allows a role from account A to assume this new role in account B"
+echo "${ACCOUNT_B_ROLE_ARN}" > "${SHARED_DIR}"/cross-account-efs-csi-driver-operator-role-arn
+
+cat << EOF > "${SHARED_DIR}"/EfsPolicyInAccountB.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeNetworkInterfaces",
+                "ec2:DescribeSubnets"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+                "elasticfilesystem:DescribeMountTargets",
+                "elasticfilesystem:DeleteAccessPoint",
+                "elasticfilesystem:ClientMount",
+                "elasticfilesystem:DescribeAccessPoints",
+                "elasticfilesystem:ClientWrite",
+                "elasticfilesystem:ClientRootAccess",
+                "elasticfilesystem:DescribeFileSystems",
+                "elasticfilesystem:CreateAccessPoint",
+                "elasticfilesystem:TagResource"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+ACCOUNT_B_POLICY=$(aws iam create-policy --policy-name "${CLUSTER_NAME}-efs-csi-policy" \
+   --policy-document file://"${SHARED_DIR}"/EfsPolicyInAccountB.json \
+   --query 'Policy.Arn' --output text) || \
+logger "INFO" "Created efs policy ${ACCOUNT_B_POLICY} in account B"
+
+
+ACCOUNT_B_ROLE_NAME="${CLUSTER_NAME}-cross-account-aws-efs-csi-operator"
+aws iam attach-role-policy \
+   --role-name "${ACCOUNT_B_ROLE_NAME}" \
+   --policy-arn "${ACCOUNT_B_POLICY}"
+logger "INFO" "Attach the Policies to the Role in account B"
+
+# STEP. In aws account A, attach an inline policy to IAM role of efs-csi-driver's controller service account 
+# with necessary permissions to perform sts assume role which created in account B.
+switch_aws_credentials default
+cat <<EOF > "${SHARED_DIR}"/AssumeRoleInlinePolicyPolicyInAccountA.json
+{
+    "Version": "2012-10-17",
+    "Statement": {
+        "Effect": "Allow",
+        "Action": "sts:AssumeRole",
+        "Resource": "${ACCOUNT_B_ROLE_ARN}"
+    }
+}
+EOF
+
+# STS standlone cluster the operator installed with specified role
+if [[ -s "${SHARED_DIR}/efs-csi-driver-operator-role-arn" ]]; then
+  EFS_CSI_DRIVER_OPERATOR_ROLE_ARN=$(cat "${SHARED_DIR}"/efs-csi-driver-operator-role-arn)
+  EFS_CSI_DRIVER_OPERATOR_ROLE="${EFS_CSI_DRIVER_OPERATOR_ROLE_ARN##*/}"
```

Comment:
If we create the policy first, we don't have to split the string like this just to get name from ARN:

```
ACCOUNT_A_POLICY_ARN=$(aws iam create-policy --policy-name "${CLUSTER_NAME}-efs-cross-account-inline-policy" \
   --policy-document file://$SCRATCH_DIR/AssumeRoleInlinePolicyPolicyInAccountA.json \
   --query 'Policy.Arn' --output text)
```

That gives us ARN and we can follow with:

```
aws iam attach-role-policy \
   --role-name "<OPERATOR_ROLE>" \
   --policy-arn "${ACCOUNT_A_POLICY_ARN}"
```

Where `<OPERATOR_ROLE>` is now `EFS_CSI_DRIVER_OPERATOR_ROLE `but possibly might be `ACCOUNT_B_ROLE_ARN` - depends on resolution of comment https://github.com/openshift/release/pull/65233/files#r2154810652

Reply:
Unfortunately `--policy-name` argument, which is mandatory when using aws iam put-role-policy. I have to revert it back.

Reply:
Sorry I went to another page previously. We use the inline policy here since the assume role policy do not need reuse to another iam role/user, so it it better that when we delete the iam role of the driver operator the policy(inline) will automatically we do not need handle the delete policy in post phase. Thanks, we could keep current way firstly and adjust if the policy needs reuse!

Reply:
Sorry, I've made a mistake. We still need the `--role-name` (and the string split). Actually using put-role-policy is shorter than what I suggested so let's keep it.

Reply:
Never mind.